parquet_file/src/catalog.rs

```diff
@@ -209,6 +217,35 @@ pub trait CatalogState {
     fn remove(&self, path: DirsAndFileName) -> Result<()>;
 }
 
+/// Find last transaction-start-timestamp.
+///
+/// This method is designed to read and verify as little as possible and should also work on most broken catalogs.
+pub async fn find_last_transaction_timestamp(
+    object_store: &ObjectStore,
+    server_id: ServerId,
+    db_name: &str,
+) -> Result<Option<DateTime<Utc>>> {
+    let mut res = None;
+    for (path, _revision_counter, _uuid) in
+        list_transaction_files(object_store, server_id, db_name).await?
+    {
+        match load_transaction_proto(object_store, &path).await {
+            Ok(proto) => match parse_timestamp(&proto.start_timestamp) {
+                Ok(ts) => {
+                    res = Some(res.map_or(ts, |res: DateTime<Utc>| res.max(ts)));
+                }
+                Err(e) => warn!("Cannot parse timestamp from {:?}: {}", path, e),
```

Comment:
```suggestion
                Err(e) => warn!(%e, ?path, "Cannot parse timestamp"),
```

If you want to use structured logging, I think that will work (print `e` with `Display` and `path` with `Debug`)